assets/wizards/popups/components/segmentation-preview/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Segmentation Preview component.
+ * Extension of WebPreview with support for "view-as-segment" functionality.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies.
+ */
+import { WebPreview } from '../../../../components/src';
+
+const SegmentationPreview = props => {
+	const postPreviewLink = window?.newspack_popups_wizard_data?.preview_post;
+	const frontendUrl = window?.newspack_popups_wizard_data?.frontend_url || '/';
+
+	const {
+		campaignGroups = [],
+		onLoad = () => {},
+		segment = '',
+		url = postPreviewLink || frontendUrl,
+	} = props;
+
+	const decorateURL = urlToDecorate => {
+		const params = {
+			view_as: [
+				`groups:${ campaignGroups.join( ',' ) }`,
+				...( segment.length ? [ `segment:${ segment }` ] : [] ),
+			].join( ';' ),
+		};
+		return addQueryArgs( urlToDecorate, params );
+	};
+
+	const onWebPreviewLoad = iframeEl => {
```

Comment:
This correctly passes the params from page to page when I navigate within the preview window. I had trouble with this working before, so this is great.